test/tools/ossfuzz/protoToAbiV2.cpp

```diff
@@ -810,13 +810,15 @@ pair<string, string> AssignCheckVisitor::visit(ArrayType const& _type)
 		length = getDynArrayLengthFromFuzz(_type.length(), counter());
 		lengthStr = to_string(length);
 		if (m_stateVar)
-			resizeBuffer = assignAndCheckStringPair(
-				m_varName + ".length",
-				m_paramName + ".length",
-				lengthStr,
-				lengthStr,
-				DataType::VALUE
-				);
+		{
+			// Dynamic storage arrays are resized via the empty push() operation
+			resizeBuffer.first = Whiskers(R"(for(uint i = 0; i < <length>; i++) <arrayRef>.push();)")
+				("length", lengthStr)
+				("arrayRef", m_varName)
+				.render();
```

Comment:
This generates the following solidity code per array dimension

```
for(uint i=0; i < length; i++) arrayRef.push();
```

where `length` is the length of the dynamically sized array, and `arrayRef` is reference to the storage array.

Example:

```
contract C {
  bool[] x;
  function f() public {
    for (uint i=0; i < 2; i++) x.push();
    x[0] = true;
    x[1] = false;
  }
}
```

Reply:
Does that also work for multi dimensional arrays? I guess shadowing on `i` is fine

Reply:
This is how code gen looks for multi-dimensional array

```
contract C {
  bool[][] x;
  function f() public {
    // Add 4 outer slots
    for(uint i=0; i<4; i++) x.push();
    // Add a 2-D array in first outer slot
    for(uint i=0; i<2; i++) x[0].push();
    // Add a 3-D array in second outer slot
    for(uint i=0; i<3; i++) x[1].push();
    ...
    x[0][0] = false;
    x[0][1] = true;
    ...
```

Is this considered shadowing on `i`?
    
  }

Reply:
Ah right, no shadowing there